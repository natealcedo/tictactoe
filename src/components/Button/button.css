.button-row {
  margin: 0 auto;
}

.button {
  border: 1px solid #c0c0c0;
  height: 50px;
  width: 300px;
  text-align: center;
  border-radius: 5px;
  background-color: #00363d;
  color: white;
  display: flex;
  margin: auto;
  margin-bottom: 5px;
}

.button:hover {
  cursor: pointer;
}

.button > div {
  margin: auto;
}