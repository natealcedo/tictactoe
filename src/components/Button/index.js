import React from "react";
import propTypes from "prop-types";
import "./button.css";

const Button = ({ label, onClick }) => (
  <div className="button" onClick={() => onClick()}>
    <div>{label}</div>
  </div>
);

Button.propTypes = {
  label: propTypes.string.isRequired,
  onClick: propTypes.func.isRequired
};

export default Button;
