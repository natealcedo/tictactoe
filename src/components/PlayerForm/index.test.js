import React from "react";
import { PlayerForm } from ".";
import { shallow } from "enzyme";
import Button from "../Button";

describe("PlayerForm", () => {
  const noOp = () => {};
  test("should render without exploding", () => {
    const wrapper = shallow(
      <PlayerForm
        isGameStarted={false}
        onChange={noOp}
        onGameStart={noOp}
        playerOneName=""
        playerTwoName=""
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test("should not call onGameStart when button is clicked and names are not filled", () => {
    const onGameStart = jest.fn();
    const wrapper = shallow(
      <PlayerForm
        isGameStarted={false}
        onChange={noOp}
        onGameStart={onGameStart}
        playerOneName=""
        playerTwoName=""
      />
    );
    wrapper.find(Button).simulate("click");
    expect(onGameStart).not.toHaveBeenCalled();
  });

  test("should call onGameStart when button is clicked", () => {
    const onGameStart = jest.fn();
    const wrapper = shallow(
      <PlayerForm
        isGameStarted={false}
        onChange={noOp}
        onGameStart={onGameStart}
        playerOneName="Nate"
        playerTwoName="James"
      />
    );
    wrapper.find(Button).simulate("click");
    expect(onGameStart).toHaveBeenCalled();
  });

  test("should call onGameStart when enter is pressed on either key inputs", () => {
    const onGameStart = jest.fn();
    const wrapper = shallow(
      <PlayerForm
        isGameStarted={false}
        onChange={noOp}
        onGameStart={onGameStart}
        playerOneName="Nate"
        playerTwoName="James"
      />
    );

    const event = {
      key: "Enter",
      keyCode: 13,
      which: 13
    };

    wrapper.find('input[name="player-one"]').simulate("keyPress", event);
    wrapper.find('input[name="player-two"]').simulate("keyPress", event);

    expect(onGameStart).toHaveBeenCalledTimes(2);
  });

  test("should not render when game is started", () => {
    const wrapper = shallow(
      <PlayerForm
        isGameStarted={true}
        onChange={noOp}
        onGameStart={noOp}
        playerOneName=""
        playerTwoName=""
      />
    );
    expect(wrapper).toBeEmptyRender();
  });

  test("should render without exploding", () => {
    const wrapper = shallow(
      <PlayerForm
        isGameStarted={false}
        onChange={noOp}
        onGameStart={noOp}
        playerOneName=""
        playerTwoName=""
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test("should call onChange with correct arguments for playerOne", () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <PlayerForm
        isGameStarted={false}
        onChange={onChange}
        onGameStart={noOp}
        playerOneName=""
        playerTwoName=""
      />
    );
    const event = {
      target: {
        value: "Jason"
      }
    };
    wrapper.find('input[name="player-one"]').simulate("change", event);
    expect(onChange).toHaveBeenCalledWith("player1", "Jason");
  });

  test("should call onChange with correct arguments for playerTwo", () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <PlayerForm
        isGameStarted={false}
        onChange={onChange}
        onGameStart={noOp}
        playerOneName=""
        playerTwoName=""
      />
    );
    const event = {
      target: {
        value: "Bourne"
      }
    };
    wrapper.find('input[name="player-two"]').simulate("change", event);
    expect(onChange).toHaveBeenCalledWith("player2", "Bourne");
  });
});
