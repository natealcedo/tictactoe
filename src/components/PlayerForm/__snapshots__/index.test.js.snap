// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerForm should render without exploding 1`] = `
<div
  className="form-container"
>
  <div
    className="input-group"
  >
    <label>
      Player one name:
    </label>
    <input
      name="player-one"
      onChange={[Function]}
      onKeyPress={[Function]}
      value=""
    />
  </div>
  <div
    className="input-group"
  >
    <label>
      Player two name:
    </label>
    <input
      name="player-two"
      onChange={[Function]}
      onKeyPress={[Function]}
      value=""
    />
  </div>
  <div
    className="button-row"
  >
    <Button
      label="Start Game"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`PlayerForm should render without exploding 2`] = `
<div
  className="form-container"
>
  <div
    className="input-group"
  >
    <label>
      Player one name:
    </label>
    <input
      name="player-one"
      onChange={[Function]}
      onKeyPress={[Function]}
      value=""
    />
  </div>
  <div
    className="input-group"
  >
    <label>
      Player two name:
    </label>
    <input
      name="player-two"
      onChange={[Function]}
      onKeyPress={[Function]}
      value=""
    />
  </div>
  <div
    className="button-row"
  >
    <Button
      label="Start Game"
      onClick={[Function]}
    />
  </div>
</div>
`;
